Marlin/src/pins/mega/pins_MEGACONTROLLER.h

```diff
@@ -129,35 +129,50 @@
 //
 // Misc. Functions
 //
-#define SDSS                                  53
+#define SDSS                             MINI_06
 #define LED_PIN                               13
 
 #ifndef CASE_LIGHT_PIN
   #define CASE_LIGHT_PIN                       2
 #endif
 
+/**
+ * MegaController LCD/SD Connector
+ *
+ *     SDD   MOSI  SDSS  --   RESET  --  LCDSS  A0    KILL  ENC
+ *     49    51    53           |         45    47    12    10
+ *     PL0   PB2   PB0          |         PL4   PL2   PB6   PB4
+ *    ----------------------------------------------------------
+ *   |  2     4     6     8    10    12    14    16    18    20 |
+ *   |  1     3     5     7     9    11    13    15    17    19 |
+ *    ----------------------------------------------------------
+ *     |     PB3   PB1    |     |         PL5   PL3   PL1   PB5
+ *     |     50    52     |     |         44    46    48    11
+ *     5V    MISO  SCK   GND   3V3   --   BL    BEEP  EN1   EN2
+ */
+#define MINI_02                               49  // SD_DETECT
+#define MINI_03                               50  // MISO
+#define MINI_04                               51  // MOSI
+#define MINI_05                               52  // SCK
+#define MINI_06                               53  // SDSS
+#define MINI_13                               44  // BACKLIGHT
+#define MINI_14                               45  // LCDSS
+#define MINI_15                               46  // BEEP
+#define MINI_16                               47  // A0
+#define MINI_17                               48  // EN1
+#define MINI_18                               12  // KILL
+#define MINI_19                               11  // EN2
+#define MINI_20                               10  // ENC
+
 //
 // LCD / Controller
 //
 #if ENABLED(MINIPANEL)
-
-  #define BEEPER_PIN                          46
-
-  #define DOGLCD_A0                           47
-  #define DOGLCD_CS                           45
-  #define LCD_BACKLIGHT_PIN                   44  // backlight LED on PA3
-
-  #define KILL_PIN                            12
-
-  #define BTN_EN1                             48
-  #define BTN_EN2                             11
-  #define BTN_ENC                             10
-
-  #define SD_DETECT_PIN                       49
-
+  // Migrated to pins/lcd
+  // Real MINIPANEL is always mapped to LCD1/2
+  #define ADAPTER_MINIPANEL_DUAL
```

Comment:
MEGACONTROLLER seems to have quite a few changes when used with MKS_MINI_12864, which enables the ADAPTER_MINIPANEL_DUAL adapter. I'm not sure which of these were actually used in this case, versus losing functionality.

- KILL_PIN is now -1, was previously 12.
- LCD_BACKLIGHT_PIN is now missing, was previously 44
- SD_DETECT_PIN is now -1, was previously 49

-

Reply:
Lots of good information in this and other comments, thanks! I'll rebase this PR shortly, then I'll aim to check all the notes and apply any needed adjustments. I don't want to keep piling more displays into this PR until what we have so far is working and not causing any pins to be defined wrongly. I'd like to merge this once it is bug-free, but I know that's a tricky thing to verify. Some of these changes actually fixed a thing or two, so that's a good sign. I do believe this is a good direction to go. I'll take all notes above under advisement!

Reply:
There may be other impacted boards as well, I haven’t gone through everything. I figured I had posted enough for the next round, and can revisit it again when ready.

Reply:
> `MEGACONTROLLER` seems to have quite a few changes when used with `MKS_MINI_12864`, which enables the `ADAPTER_MINIPANEL_DUAL`

Oops. Forgot to include in `pins_lcd.h`…
```cpp
  #elif ENABLED(MINIPANEL)
    #include "lcd/MINIPANEL.h"
```
Everything else looks correct, but I will triple-check.